StreamerBotPlugin/Commands/ActionAdjustmentCommand.cs

```diff
@@ -34,27 +36,41 @@ public class ActionAdjustmentCommand : PluginDynamicAdjustment
         private readonly HttpService _httpService;
         private Action[] _actions;
 
+        private static readonly ConcurrentDictionary<String, Int32> _ActionAdjustmentValues = new ConcurrentDictionary<String, Int32>();
+
         public ActionAdjustmentCommand() : base("Adjustment Action", "Choose and execute your actions with adjustments (Reset sets it to 0)", "Adjustment Action", true)
         {
             this._httpService = HttpService.Instance;
-            this._actions = this._httpService.GetActions()?.Actions ?? Array.Empty<Action>();
+            this._setActions().Wait();
             this.MakeProfileAction("tree");
         }
 
+        private async Task<Action[]> _setActions() =>
+            this._actions = (await this._httpService.GetActions()).Actions;
+
+
         protected override PluginProfileActionData GetProfileActionData()
         {
-            this._actions = this._httpService.GetActions()?.Actions ?? Array.Empty<Action>();
+            this._setActions().Wait();
             var tree = new PluginProfileActionTree("Select Windows Settings Application");
 
             tree.AddLevel("Category");
             tree.AddLevel("Command");
 
-            var node = tree.Root.AddNode("Actions");
-
-            foreach (var item in this._actions.Where(action => action.Enabled))
+            this._actions.GroupBy(x => x.Group).ToList().ForEach(actionGroup =>
             {
-                node.AddItem(item.Id, item.Name, $"Execute {item.Name}");
-            }
+                var node = tree.Root.AddNode(actionGroup.Key);
+
+                foreach (var item in actionGroup.Where(action => action.Enabled))
+                {
+                    if (!_ActionAdjustmentValues.ContainsKey(item.Id))
+                    {
+                        _ActionAdjustmentValues.TryAdd(item.Id, 0);
+                    }
+
+                    node.AddItem(item.Id, item.Name, $"Execute {item.Name}");
+                }
+            });
 
```

Comment:
I like this Idea! Thank you!

Reply:
:D you're welcome. 
Initially, I switched it to `list` and had some success with that, but realized it needed the extra levels provided by `tree`.